### ISSUE TITLE
valid-lang should fail lang=" "

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2456]
Lang attributes with whitespace cause problems in screen readers. Because of that, the `valid-lang` rule should fail the following code snippet:

```html
<article lang=" ">
  The quick brown fox jumped over the lazy dog
</article>
```

Because updating the has-lang check to fail with whitespace-only content, we'll need to change the `has-lang` check to not fail when it is passed whitespace text.

**Comment by xyc** [comment]
@WilcoFiers Should `valid-lang` also fail with an empty string? (`lang=""`) It seems to allow empty string because of what happens on Edge.

https://github.com/dequelabs/axe-core/blob/32888d43d72bc2484e2c00b5f00fb27d92ac62b5/lib/checks/language/valid-lang-evaluate.js#L15-L19

**Comment by padmavemulapati** [comment]
valid-lang is failing only when lang has whitespace value
test data: `<div id="target" lang=" ">hello</div>`
 
![image](https://user-images.githubusercontent.com/47409579/93748242-89861500-fc15-11ea-9595-a6280f466174.png)

**Comment by straker** [comment]
No docs required
